File: typescript/easy/largets-3-same-digit-number-in-string.ts
function largestGoodInteger(num: string): string {
    let count = 1;
    let ans = -1;

    for (let i = 1; i < num.length; i++) {
        if (num[i] === num[i - 1]) {
            count++;
            if (count === 3) {
                ans = Math.max(ans, parseInt(num[i]));
            }
        } else {
            count = 1;
        }
    }

    return ans !== -1 ? String(ans).repeat(3) : "";
}

File: csharp/easy/largets-3-same-digit-number-in-string.cs
string LargestGoodInteger(string num)
    {
        int count = 1;
        int ans = -1;

        for (int i = 1; i < num.Length; i++)
        {
            if (num[i] == num[i - 1])
            {
                count++;
                if (count == 3)
                {
                    ans = Math.Max(ans, num[i] - '0');
                }
            }
            else
            {
                count = 1;
            }
        }

        return ans != -1 ? new string((char)('0' + ans), 3) : "";
    }
